src/test/sh/Plugin.Testing.Stub-1.0.0/usr/opt/EXAplugins/Testing.Stub-1.0.0/exaoperation-gate/plugin-functions

```diff
@@ -1,32 +1,57 @@
-#!/usr/bin/env python3
-import sys, os, base64
-backup_home = "/opt/testing-stub"
-cfg_file = "%s/sync.cfg" % backup_home
-key_file = "%s/key" % backup_home
-
-
-# definitions - interfaces
-plugin_functions = {
-    'INSTALL': 'Install plugin.',
-    'UNINSTALL': 'Install plugin.',
-    'STATUS': 'Show status of plugin (not installed, activated, not activated).',
-    'START': 'Start plugin service',
-    'STOP': 'Stop plugin service',
-    'RESTART': 'Restart the service',
-    'ON_BOOT': 'Undocumented'
+#!/bin/bash
+
+# This script is a replacement for the plugin-functions script that comes with
+# a regular Exasol installation. It uses Bash instead of Python to reduce the
+# number of external dependencies.
+#
+# Output is identical to the original script in order to not break potential
+# automation that might parse this output.
+
+readonly script_path="$(dirname $(readlink -f $0))"
+
+declare -A functions
+
+functions[INSTALL]='Install plugin.'
+functions[UNINSTALL]='Install plugin.'
+functions[STATUS]='Show status of plugin (not installed, activated, not activated).'
+functions[START]='Start plugin service'
+functions[STOP]='Stop plugin service'
+functions[RESTART]='Restart the service'
+functions[ON_BOOT]='Undocumented'
+
+usage() {
+    echo "Usage: $0 <function name> <parameter>"
+    echo "       $0 --show-functions"
+    echo "       $0 -h | --help"
+}
+
+exit_abnormal() {
+    echo 'Wrong call to plugin wrapper.'
+    exit 1
+}
+
+list_functions() {
+    for key in "${!functions[@]}"; do
+        echo "$key:${functions[$key]}"
+    done
 }
 
-# implement show-functions
-if len(sys.argv) == 2 and sys.argv[1] == '--show-functions':
-    print("\n".join(["%s:%s" % (X, Y) for X, Y in plugin_functions.items()]))
-    sys.exit(0)
-elif len(sys.argv) != 3:
-    print("Wrong call to plugin wrapper.")
-    sys.exit(1)
-elif sys.argv[1].upper() not in plugin_functions:
-    print("Unknown function '%s'" % sys.argv[1])
-    sys.exit(1)
-
-# call actual function
-binary = "%s/%s" % (os.path.dirname(os.path.abspath(__file__)), sys.argv[1].lower())
-os.execl(binary, binary, *sys.argv[2:])
\ No newline at end of file
+if [[ "$#" -eq 1 ]]; then
+    if [[ "$1" == '--show-functions' ]]; then
+        list_functions
+    elif [[ "$1" == "-h" ]] || [[ "$1" == "--help" ]]; then
```

Comment:
On real EXAoperation, this can never happen: The only call done without second argument is "--show-functions"

Reply:
Users wanting to know what the script does can still call it from the command line (e.g. on an ssh shell).

Reply:
🤷‍♂